lib/utils/socket-config.js

```diff
@@ -0,0 +1,66 @@
+import { readFile } from 'node:fs/promises'
+
+import Ajv from 'ajv'
+import { ErrorWithCause } from 'pony-cause'
+import { parse as yamlParse } from 'yaml'
+
+import { isErrnoException } from './type-helpers.js'
+
+/**
+ * @typedef SocketYml
+ * @property {2} version
+ * @property {string[]} [projectIgnorePaths]
+ * @property {{ [issueName: string]: boolean }} [issueRules]
+ */
+
+/** @type {import('ajv').JSONSchemaType<SocketYml>} */
+const socketYmlSchema = {
+  $schema: 'http://json-schema.org/draft-07/schema#',
+  type: 'object',
+  properties: {
+    version: { type: 'integer' },
+    projectIgnorePaths: {
+      type: 'array',
+      items: { type: 'string' },
+      nullable: true,
+    },
+    issueRules: {
+      type: 'object',
+      additionalProperties: { type: 'boolean' },
+      nullable: true,
+      required: [],
+    },
+  },
+  required: ['version'],
+  additionalProperties: true,
+}
+
+/**
+ * @param {string} filePath
+ * @returns {Promise<SocketYml|undefined>}
+ */
+export async function readSocketConfig (filePath) {
```

Comment:
Should break this function up into 2 steps when we modularize this. A function that parses and validates string content, and another that wraps the first function that includes the file io step.

Reply:
Yep, already done that in my extracted version 🙂

Reply:
👌